webui/ocw/lib/EC2.py

```diff
@@ -39,27 +27,32 @@ def check_credentials(self):
                 time.sleep(1)
         raise Exception("Invalid EC2 credentials")
 
-    def ec2_resource(self, region='eu-central-1'):
-        if region not in self.__ec2_resource:
-            self.__ec2_resource[region] = boto3.resource('ec2', aws_access_key_id=self.__key,
-                                                         aws_secret_access_key=self.__secret,
+    def get_resource_region(self, region='eu-central-1'):
+        if region not in self.__resource_regions or self.__credentials.isExpired():
+            key, secret = self.getKey_and_secret()
```

Comment:
I think it will not work. The `isExpired()` will return only once `True` and then you are calling `getData()` which lead to new credentials and new expire date.
A call later call to `get_client_region()` will not recreate the `self.__client_regions` with new key and pw, or?